wagtail/api/v2/pagination.py

```diff
@@ -33,7 +33,7 @@ def paginate_queryset(self, queryset, request, view=None):
         stop = offset + limit
 
         self.view = view
-        self.total_count = queryset.count()
+        self.total_count = len(queryset) if isinstance(queryset, list) else queryset.count()
```

Comment:
Of course ; I've just updated my previous commit. :wink: It turns out that `queryset` can be an instance of `QuerySet` but it can also be an instance of `BaseSearchResults` (which is the case if you use the Elasticsearch backend and the `search` parameter for example).